CHANGELOG.md

```diff
@@ -1,3 +1,13 @@
+## 0.12.1
+
+### New Features
+
+* Adds a new return type for `IOClient` that includes the ability to detach a 
+  socket. To use or reference `IOStreamedResponse` directly, you will need to 
+  import the new `package:http/io_streamed_response.dart` import. Then cast 
+  a `StreamedResponse` as an `IOStreamedReesponse` to call the `detachSocket()`
+  method. 
+
 ## 0.12.1-dev
 
 * Remove dependency on `package:async`.
```

Comment:
If we change the return type we won't need to ask people to cast. If the client they are using is statically an `IOClient` then the response will also be statically an `IOStreamedResponse`. No cast should be necessary.

Also, the imports mentioned here don't line up with the implementation. Finally, I don't think we need the changelog section.

```suggestion
## 0.12.1-dev

* Add `IOStreamedResponse` which includes the ability to detach the socket.
  When sending a request with an `IOClient` the response will be an
  `IOStreamedResponse`.
* Remove dependency on `package:async`.
```

Reply:
👍 This is much cleaner, thanks! Updated in b699b87.